### PR TITLE
[Goals]: Add a long term goal template

### PR DESCRIPTION
**Pull request by youngcw**
This adds in the ability to set a goal in a strict sense.  Instead of a monthly auto budget template like what we have had up till now, this adds in the option to set a purely visual goal.  This type wont apply anything to the budget but will flag the goal as a "long goal" in the db that is then used to color the balance for personal reference.

For a basic test use a note like `#goal 500`.  Then try out different budget amounts to adjust the balance.  If the total balance is <500 it should be marked as unmet.  If >=500, met.

The goal directive will not unset the budgeted amount, so the category acts like a normal category, just with a colored balance.

This should work in conjunction with other templates too.  For example
```
#template 100 up to 500
#goal 500
```
The goals get run after templates so the value to get green coloring is 500 not 100, but 100 will automatically get budgeted.  In this case the budgeted amount will get overwritten since there is a template in the category.

Based on #3011
docs: actualbudget/docs#401

### CODE REVIEW

**Comment by youngcw** [comment]
@gschmidl Should be fixed now

**Comment by atgrey24** [comment]
The "Apply Budget Template"  option for an individual category doesn't seem to be working on individual categories. Behavior works as expected for both Apply and Overwrite on the whole budget month.

Overall I really like this new capability, and think it should be included in the eventual goals UI #496

File: packages/loot-core/src/server/aql/schema/index.ts [hunk 1]
```diff
@@ -167,6 +168,7 @@ export const schema = {
     amount: f('integer'),
     carryover: f('integer'),
     goal: f('integer'),
+    long_foal: f('integer'),
```
Comment (shall0pass) [inline packages/loot-core/src/server/aql/schema/index.ts:171]: long_goal:

**Comment by youngcw** [comment]
@atgrey24 I think its all working now.  Can you test it again?

**Comment by atgrey24** [comment]
> @atgrey24 I think its all working now. Can you test it again?

Had to reset the budget cache, but now it seems to be working as expected. Thanks!

**Review by matt-fidd — APPROVED** [review]
Nothing pops out in the code and functionally it's handled everything I've thrown at it.
Looking forward to using this!

**Pull request merged**